msm/src/serialization/mod.rs

```diff
@@ -44,6 +48,40 @@ mod tests {
         BaseSponge, Fp, OpeningProof, ScalarSponge, BN254, N_LIMBS,
     };
 
+    use ark_ff::{FftField, Field, PrimeField};
+
+    impl LookupTable {
+        fn into_lookup_vector<F: FftField + PrimeField + Field>(
+            self,
+            domain: EvaluationDomains<F>,
+        ) -> Vec<Lookup<F>> {
+            assert!(domain.d1.size >= (1 << 15));
+            match self {
+                Self::RangeCheck15 => (0..(1 << 15))
+                    .map(|i| Lookup {
+                        table_id: LookupTable::RangeCheck15,
+                        numerator: -F::one(),
```

Comment:
Why are these `-1` by default? I'd assume that by default we are not using any of the elements, so they should have `0`?

Reply:
See equation (15): https://eprint.iacr.org/2022/1530.pdf.

Reply:
Ah, ok, I think I misunderstood what this function does. Please tell me if I'm right: what you're doing is given a fixed table you're building a set of queries for that table covering every element of this table?